pkg/ingestor/key/key_test.go

```diff
@@ -297,28 +310,28 @@ func TestDelete(t *testing.T) {
 }
 
 type mockKeyProvider struct {
-	collector map[string]Key
+	collector map[string]*Key
```

Comment:
hmm quick question on the change here? is this related to ctx change or just a drive by?

Reply:
So as I was adding in the context, the unit test started to fail. Some keys were being overwritten before the test could complete. That was the reason for the changes. Added issue #232 to capture this.

Reply:
awesome - tks!